parametric_plasma_source/plasma.py

```diff
@@ -0,0 +1,380 @@
+        else:
+            self._single_null = single_null
+
+    @property
+    def minor_radius(self):
+        return self._minor_radius
+
+    @minor_radius.setter
+    def minor_radius(self, minor_radius):
+        self._minor_radius = minor_radius
+
+    @property
+    def major_radius(self):
+        return self._major_radius
+
+    @major_radius.setter
+    def major_radius(self, major_radius):
+        if major_radius > 2000 or major_radius < 1:
+            raise ValueError('major_radius is out of range')
+        else:
+            self._major_radius = major_radius
+
+    @property
+    def elongation(self):
+        return self._elongation
+
+    @elongation.setter
+    def elongation(self, elongation):
+        if elongation > 4 or elongation < 0:
+            raise ValueError('elongation is out of range')
+        else:
+            self._elongation = elongation
+
+    @property
+    def ion_density_pedistal(self):
+        return self._ion_density_pedistal
+
+    @ion_density_pedistal.setter
+    def ion_density_pedistal(self, ion_density_pedistal):
+        if ion_density_pedistal > 10e22 or ion_density_pedistal < 1e4:
+            raise ValueError('ion_density_pedistal is out of range')
+        else:
+            self._ion_density_pedistal = ion_density_pedistal
+
+    def replace_variable_value(
+        self, input_strings, variable_name, new_value, is_cpp=True
+    ):
+        """Replaces the value assigned to a variable with the provided value
+
+        param: input_strings: The list of strings to search for the variable.
+        type input_strings: List[str]
+        
+        param: variable_name: The name of the variable to search for.
+        type variable_name: str
+        
+        param: new_value: The value to set the variable to.
+        type new_value: Union[str, float, int]
+
+        param is_cpp: Whether the variable is a C++ variable.
+                      Optional, by default True.
+        type is_cpp: bool
+
+        raises ValueError: variable_name was not found in input_strings.
+        """
+        if is_cpp:
+            strs_to_find = ("const", variable_name, "=", ";")
+        else:
+            strs_to_find = (variable_name, "=")
+        for idx, string in enumerate(input_strings):
+            if all(str_to_find in string for str_to_find in strs_to_find):
+                equals_idx = string.find("=")
+                if equals_idx >= 0:
+                    if is_cpp:
+                        new_value = f"{new_value};"
+                    input_strings[idx] = string.replace(
+                        input_strings[idx][equals_idx:], f" = {new_value}"
+                    )
+                    break
+        else:
+            file_content = "\n".join(input_strings)
+            raise ValueError(
+                f"{variable_name} string not found in {file_content}"
+            )
+
+    def export_plasma_source(self, output_filename):
```

Comment:
can you confirm if this is needed? If the pybinding is done right, there isnt a need for this anymore AFAIK one should be able to set the values with the exposed setters, can yourself or @DanShort12 confirm?

Reply:
I had bit of a think while writing the bindings and I think there is still the use case where the shared object would need to be generated from the Python. Since OpenMC reads the shared object rather than the Python instance as part of the source sampling routine, I think each run needs its own source sampling shared object.

Imagine also the use case where you have two Python instances generated from the same `.so` and then run OpenMC - how would the shared object loaded into OpenMC know the parameters to use? I think in that case both the instances would need to generate their own `.so` and OpenMC would need to be told which one to use.

I can do a bit more thinking/Googling, or I might have misunderstood something, so happy to be corrected.

Reply:
Thinking about this a bit more, I wonder if the set up of the plasma source could be done by serialising the state of the parameterisation into XML and adding that to the settings.xml. That would need an update in OpenMC but could be a cleaner solution that having to share memory between python and c++ (and means the provision of the parameterisation isn't language dependent). Will see if I can knock something together.